cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -144,6 +197,71 @@ protected void runOneIteration() throws Exception {
   }
 
   /**
+   * Perform aggregation on the Streams described by the {@code streamNames}, and no other Streams.
+   * If aggregation was previously done on other Streams, those must be cancelled.
+   *
+   * @param streamNames names of the streams to perform data sizes aggregation on
+   */
+  public void aggregate(Set<String> streamNames) {
+    Set<String> existingAggregators = Sets.newHashSet(aggregators.keySet());
+    for (String streamName : streamNames) {
+      if (existingAggregators.remove(streamName)) {
+        continue;
+      }
+
+      long filesSize = 0;
+      try {
+        StreamConfig config = streamAdmin.getConfig(streamName);
+        filesSize = StreamUtils.fetchStreamFilesSize(config);
+      } catch (IOException e) {
+        LOG.error("Could not compute sizes of files for stream {}", streamName);
+      }
+
+      Aggregator aggregator = new Aggregator(streamName, filesSize);
+      aggregator.startAndWait();
+      aggregators.put(streamName, aggregator);
+    }
+
+    // Stop aggregating the heartbeats we used to listen to before the call to that method,
+    // but don't anymore
+    for (String outdatedStream : existingAggregators) {
+      Aggregator aggregator = aggregators.remove(outdatedStream);
+      aggregator.stopAndWait();
```

Comment:
Actually, that's because we don't iterator through ``existingAggregators`` in the first for-loop, the logic is a little more twisted. We iterator through ``streamNames``, and want to remove from ``existingAggregators`` all the streams that were not found in ``streamNames``.